ui/src/components/Secrets.vue

```diff
@@ -517,6 +577,75 @@ function removeSecret(counter) {
     secrets.value[0] = { key: "", value: "", file: [] }
   }
 }
+
+function generateSecureRandomString(length, includeUppercase, includeLowercase, includeNumbers, includeSpecial) {
+  const classes = {
+    upperCase: { chars: "ABCDEFGHIJKLMNOPQRSTUVWXYZ", enabled: includeUppercase },
+    lowerCase: { chars: "abcdefghijklmnopqrstuvwxyz", enabled: includeLowercase },
+    digits: { chars: "0123456789", enabled: includeNumbers },
+    special: { chars: specialCharacters.value, enabled: includeSpecial }
+  };
+
+  const chars = Object.values(classes)
+    .filter(v => v.enabled)
+    .map(v => v.chars)
+    .join('');
+
+  if (chars.length === 0) {
+    setErrorMessage('At least one character type must be selected to create a random string.');
+    return '';
+  }
+
+  let initialResult = '';
+
+  Object.values(classes).forEach(v => {
+    if (v.enabled) {
+      initialResult += getRandomCharacter(v.chars);
+    }
+  });
+  for (let i = initialResult.length; i < length; i++) {
+    initialResult += getRandomCharacter(chars);
+  }
+  const result = shuffleString(initialResult);
+  return result;
+}
+
+function shuffleString(str) {
+  const array = str.split('');
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+  return array.join('');
+}
+
+function getRandomCharacter(charSet) {
```

Comment:
This is probably better, than the last implementation.
Another approach would be to use the random bits and ignore those random values, that are outside the range of our chars:
```javascript
function getRandomCharacter(charSet) {
  const randomValues = new Uint32Array(1);
  const needBits = Math.ceil(Math.log2(charSet.length))
  if (needBits>30) {
    setErrorMessage("charSet too big")
    return '';
  }
  const mask = (1 << needBits) - 1
  for (let counter=0; counter < 20; counter++) {
    window.crypto.getRandomValues(randomValues);
    const randomIndex = randomValues[0] & mask;
    if (randomIndex < charSet.length) {
      return charSet[randomIndex];
    }
  }
  setErrorMessage("Could not generate a random char within the given loop limit")
  return ''
}
```

This would have a problem with the loop, as it could generate empty chars, which would have to be catched later by the outer function.
On the other hand, this doesn't use floating points, so has no problems with rounding errors (which should not really be a problem in both implementations).